crates/proc_macro_api/src/msg.rs

```diff
@@ -77,13 +77,24 @@ impl Message for Request {}
 impl Message for Response {}
 
 fn read_json(inp: &mut impl BufRead) -> io::Result<Option<String>> {
-    let mut buf = String::new();
-    inp.read_line(&mut buf)?;
-    buf.pop(); // Remove trailing '\n'
-    Ok(match buf.len() {
-        0 => None,
-        _ => Some(buf),
-    })
+    loop {
+        let mut buf = String::new();
```

Comment:
I mean, we _could_ pass this string in to reuse the storage, but maybe we don't make enough of them?

It's 52 MB in total for RA.